### PR TITLE
ci: speed up prettier/eslint pre-commit hooks and use types instead of files

### PR DESCRIPTION
**Pull request by thiagowfx**
c.f. https://github.com/pre-commit/mirrors-prettier/blob/main/.pre-commit-hooks.yaml and https://github.com/pre-commit/identify/blob/main/identify/extensions.py

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -108,26 +108,24 @@ repos:
...
         language: system
         verbose: true
-        files: \.[jt]s$
-        types: [file]
+        types: [javascript, ts]
       # ESLint may update code style (formatting) so keep prettier last.
       - id: prettier
         name: prettier
-        entry: 'npx prettier --check .'
+        entry: 'npx prettier --write --ignore-unknown --check .'
```
Comment (Lightning00Blade) [inline .pre-commit-config.yaml:129]: Do we need the `.` at the end? I think that will run all files Prettier can format in the repo? We may if we want to format `.yml`, `.json` etc.

**Review by Lightning00Blade — COMMENTED** [review]
Please verify that EsLint is running only once as I had issue when it ran one time per file changed

File: .pre-commit-config.yaml (hunk 1 above)
Comment (thiagowfx) [reply .pre-commit-config.yaml]: No. Removed.

Comment (thiagowfx) [reply .pre-commit-config.yaml]: Yes, that's OK. Markdown, YAML, etc. Whatever prettier supports.

**Comment by thiagowfx** [comment]
I had made a mistake. It's `types_or`, not `types`. Fixed.

**Review by Lightning00Blade — APPROVED** [review]

**Pull request merged**